Deck_Shuffler/Deck_Shuffler/Shuffler.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Deck_Shuffler
+{
+    class Shuffler
+    {
+        public static Deck Shuffle(Deck DeckToShuffle)
+        {
+            Random rnd = new Random();
+            List<int> indexes = Enumerable.Range(0, 52).ToList();
+            //var randomisedIndexes = indexes.OrderBy(idx => rnd.Next());
+
+            List<int> randomisedIndexes = new List<int>();
+            while (indexes.Count>0)
+            {
+                int randomNo = rnd.Next(0,52);
```

Comment:
This algorithm looks a little bit inefficient. You're basically getting a random number from 0 to 52 even if you've already removed 51 cards from indexes! Might take 52 (or more) turns round the while loop before it get's lucky on the last one. I admit, I'm not sure how I'd do it any better without looking on the internet (or sitting down and thinking about it for a while!), but wondering if you could think of anything more efficient if possible...